assets/js/blocks/checkout/inner-blocks/checkout-billing-address-block/block.tsx

```diff
@@ -91,7 +91,14 @@ const Block = ( {
 
 	return (
 		<AddressFormWrapperComponent>
-			<StoreNoticesContainer context={ noticeContexts.BILLING_ADDRESS } />
+			<StoreNoticesContainer
+				context={ noticeContexts.BILLING_ADDRESS }
+				capturedContexts={
+					useBillingAsShipping
+						? [ noticeContexts.SHIPPING_ADDRESS ]
+						: []
+				}
```

Comment:
I don't feel strongly about approach, another option would be to call `registerContainer` from this block but I didn't test that yet.